File: Visual Studio Product Team/Visual Studio 2013 SDK Samples/115802-Visual Studio 2013 SDK Samples/Ook Language sample - VS 2013/VB/Classification/ClassificationType.vb
Imports System.ComponentModel.Composition
Imports Microsoft.VisualStudio.Text.Classification
Imports Microsoft.VisualStudio.Utilities

Namespace OokLanguage

    Friend NotInheritable Class OrdinaryClassificationDefinition

        ''' <summary>
        ''' Defines the "ordinary" classification type.
        ''' </summary>
        <Export(GetType(ClassificationTypeDefinition)), Name("ook!")>
        Friend Shared ookExclamation As ClassificationTypeDefinition = Nothing

        ''' <summary>
        ''' Defines the "ordinary" classification type.
        ''' </summary>
        <Export(GetType(ClassificationTypeDefinition)), Name("ook?")>
        Friend Shared ookQuestion As ClassificationTypeDefinition = Nothing

        ''' <summary>
        ''' Defines the "ordinary" classification type.
        ''' </summary>
        <Export(GetType(ClassificationTypeDefinition)), Name("ook.")>
        Friend Shared ookPeriod As ClassificationTypeDefinition = Nothing

    End Class
End Namespace

File: Visual Studio Product Team/Visual Studio 2013 SDK Samples/115802-Visual Studio 2013 SDK Samples/Ook Language sample - VS 2013/C#/Classification/ClassificationType.cs
using System.ComponentModel.Composition;
using Microsoft.VisualStudio.Text.Classification;
using Microsoft.VisualStudio.Utilities;

namespace OokLanguage
{
    internal static class OrdinaryClassificationDefinition
    {
        #region Type definition

        /// <summary>
        /// Defines the "ordinary" classification type.
        /// </summary>
        [Export(typeof(ClassificationTypeDefinition))]
        [Name("ook!")]
        internal static ClassificationTypeDefinition ookExclamation = null;

        /// <summary>
        /// Defines the "ordinary" classification type.
        /// </summary>
        [Export(typeof(ClassificationTypeDefinition))]
        [Name("ook?")]
        internal static ClassificationTypeDefinition ookQuestion = null;

        /// <summary>
        /// Defines the "ordinary" classification type.
        /// </summary>
        [Export(typeof(ClassificationTypeDefinition))]
        [Name("ook.")]
        internal static ClassificationTypeDefinition ookPeriod = null;

        #endregion
    }
}
